service/src/main/java/com/epam/esm/service/OrderService.java

```diff
@@ -0,0 +1,127 @@
+import com.epam.esm.repository.exception.InvalidPageContextException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.List;
+
+import static java.time.ZoneOffset.UTC;
+
+/**
+ * This service class encapsulated business logic related to {@link Order} entity.
+ *
+ * @author Dmitry Karnyshov
+ */
+@Service
+public class OrderService {
+    private OrderRepository orderRepository;
+    private UserRepository userRepository;
+    private GiftCertificateRepository certificateRepository;
+
+    public OrderService(
+            OrderRepository orderRepository,
+            UserRepository userRepository,
+            GiftCertificateRepository certificateRepository) {
+        this.orderRepository = orderRepository;
+        this.userRepository = userRepository;
+        this.certificateRepository = certificateRepository;
+    }
+
+    /**
+     * Retrieve all orders.
+     *
+     * @param pageContext {@link PageContext} object with pagination logic
+     * @throws InvalidPageContextException if passed page or page size values are invalid
+     * @return list of {@link OrderDto}
+     */
+    public List<OrderDto> findAll(PageContext pageContext) {
+        return orderRepository.findAll(pageContext)
+                .stream()
+                .map(OrderDto::fromOrder)
+                .toList();
+    }
+
+
+    /**
+     * Retrieve all orders of specified user.
+     *
+     * @param userId user id
+     * @param pageContext {@link PageContext} object with pagination logic
+     * @throws InvalidPageContextException if passed page or page size values are invalid
+     * @return list of {@link OrderDto}
+     */
+    public List<OrderDto> findByUser(long userId, PageContext pageContext) {
+        return orderRepository.findByUser(pageContext, userId)
+                .stream()
+                .map(OrderDto::fromOrder)
+                .toList();
+    }
+
+    /**
+     * Retrieve order by its unique id.
+     *
+     * @param id order id
+     * @throws EntityNotFoundException in case when order with this id does not exist
+     * @return {@link TagDto} object
+     */
+    public OrderDto findById(long id) {
+        Order order = orderRepository.findById(id)
+                .orElseThrow(() -> new EntityNotFoundException(id, Order.class));
+        return OrderDto.fromOrder(order);
+    }
+
+    /**
+     * Make an order.
+     *
+     * @param orderDto {@link OrderDto} instance (only {@code userId} and {@code certificateId} are required)
+     * @throws EmptyOrderException in case when passed list of certificate ids is empty
+     * @throws EntityNotFoundException in case when user or (and) certificate with specified ids do not exist
+     * @return {@link OrderDto} object that represents created order
+     */
+    @Transactional
+    public OrderDto makeOrder(OrderDto orderDto) {
```

Comment:
The method is a bit too long. Let's separate it. For example we can move out validation, order creation, smth else perhaps

Reply:
Fixed.